vault/server.sls

```diff
@@ -21,8 +21,7 @@ generate self signed SSL certs:
     - user: root
     - group: root
     - mode: 755
-
-{%- if vault.dev_mode %}
```

Comment:
This actually makes me realize that we need a superset of `service` configuration. 
It doesn't make sense to describe it as `dev_mode` but it does make sense to describe it as a service.

Ie. you don't need `/etc/vault/config` if you just want the vault CLI but you do want it if you have a service (dev mode is just one mode).

So i say keep this, or add to the pillar:

```
vault:
  service:
    enabled: True
    dev_mode: True
```

and use this logic here instead of just deciding on `vault.dev_mode` being true/false